### PR TITLE
Update CONNECT-IP to match the change to HTTP datagrams

### PR DESCRIPTION
**Pull request by tfpauly**
Aligning CONNECT-IP with the changes in HTTP datagrams and CONNECT-UDP.

### CODE REVIEW

**Review by mirjak — APPROVED** [review]

**Review by DavidSchinazi — APPROVED** [review]
Overall looks great, thanks for writing this!
Approved modulo 2 small comments, (and also please don't merge until the corresponding HTTP Datagrams PR is merged)

File: draft-ietf-masque-connect-ip.md [hunk 1]
```diff
@@ -118,9 +118,8 @@ When using HTTP/2 or HTTP/3, the following requirements apply to requests:
   determine the scope of the request, such as requesting full-tunnel IP packet
   forwarding, or a specific proxied flow, see {{scope}}.
 
-Along with a request, the client can send a REGISTER_DATAGRAM_CONTEXT capsule
-{{HTTP-DGRAM}} to negotiate support for sending IP packets in HTTP Datagrams
-({{packet-handling}}).
+The client also includes the "Capsule-Protocol" header with a value of "?1"
```
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-ietf-masque-connect-ip.md:121]: Yeah, SHOULD makes sense

File: draft-ietf-masque-connect-ip.md [hunk 2]
```diff
@@ -340,36 +339,70 @@ in the same ROUTE_ADVERTISEMENT capsule, they MUST follow these requirements:
...
+# Context Identifiers
+
+This protocol allows future extensions to exchange HTTP Datagrams which carry
+different semantics from IP packets. For example, an extension could define
+a way to send compressed IP header fields. In order to allow for this
+extensibility, all HTTP Datagrams associated with IP proxying request streams
+start with a context ID, see {{payload-format}}.
+
+Context IDs are 62-bit integers (0 to 2<sup>62</sup>-1). Context IDs are encoded
+as variable-length integers, see {{Section 16 of !QUIC=RFC9000}}. The context ID
```
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-ietf-masque-connect-ip.md:351]: It does work!

https://ietf-wg-masque.github.io/draft-ietf-masque-connect-ip/tfp/datagram-update/draft-ietf-masque-connect-ip.html

File: draft-ietf-masque-connect-ip.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tfpauly) [reply draft-ietf-masque-connect-ip.md:121]:
```suggestion
The client SHOULD also include the "Capsule-Protocol" header with a value of "?1"
```

**Comment by tfpauly** [comment]
@DavidSchinazi when do you plan to merge the other PRs?

File: draft-ietf-masque-connect-ip.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (DavidSchinazi) [reply draft-ietf-masque-connect-ip.md:351]: Sweet! I guess I need to update my local software :P

**Comment by DavidSchinazi** [comment]
@tfpauly I plan on merging the other PRs as soon as we have WG consensus, which (assuming no objections) should happen at the end of the month (see [email from chairs](https://mailarchive.ietf.org/arch/msg/masque/2skE5g0wH4EMAzPm6m3lH3Fo_cc/))

**Pull request merged**